lib/tracker/handlers/base.rb

```diff
@@ -52,6 +54,17 @@ def params
     CGI.parse(env["QUERY_STRING"] || "").deep_symbolize_keys
   end
 
+  def request_params
+    return {} unless env.present?
+
+    rack_input = env['rack.input'].read
```

Comment:
you'll have to rewind here again in ensure block just in case...https://content.pivotal.io/blog/sanitizing-post-params-with-custom-rack-middleware